lib/AST/ModuleNameLookup.cpp

```diff
@@ -118,130 +214,147 @@ static ResolutionKind recordImportDecls(LazyResolver *typeResolver,
 
     // Update the overload set.
     bool stillOverloadable = updateOverloadSet(overloads, newDecls);
-    return stillOverloadable ? ResolutionKind::Overloadable
-                             : ResolutionKind::Exact;
+    if (!stillOverloadable)
+      resolutionKind = ResolutionKind::Exact;
+    break;
   }
 
   case ResolutionKind::Exact:
     // Add all decls. If they're ambiguous, they're ambiguous.
     results.append(newDecls.begin(), newDecls.end());
-    return ResolutionKind::Exact;
+    break;
 
   case ResolutionKind::TypesOnly:
     // Add type decls only. If they're ambiguous, they're ambiguous.
-    std::copy_if(newDecls.begin(), newDecls.end(), std::back_inserter(results),
-                 [](const ValueDecl *VD) { return isa<TypeDecl>(VD); });
-    return ResolutionKind::TypesOnly;
+    llvm::copy_if(newDecls, std::back_inserter(results),
+                  [](const ValueDecl *VD) { return isa<TypeDecl>(VD); });
+    break;
+  }
+
+  return results.size() != originalSize;
+}
+
+template <typename OverloadSetTy>
+void ModuleNameLookup<OverloadSetTy>::collectLookupResultsFromImports(
+    ArrayRef<ModuleDecl::ImportedModule> reexports,
+    ModuleDecl::AccessPathTy accessPath,
+    SmallVectorImpl<ValueDecl *> &decls,
```

Comment:
Is `decls` an out parameter?

Reply:
Yep. Does it need something explicit beyond being a mutable reference?